edgelet/iotedged/src/lib.rs

```diff
@@ -384,16 +386,46 @@ where
 
         match settings.provisioning() {
             Provisioning::Manual(manual) => {
-                info!("Starting provisioning edge device via manual mode...");
-                let (key_store, provisioning_result, root_key) =
-                    manual_provision(&manual, &mut tokio_runtime)?;
-                start_edgelet!(
-                    key_store,
-                    provisioning_result,
-                    root_key,
-                    force_module_reprovision,
-                    None,
-                );
+                match manual.authentication_method() {
+                    ManualAuthMethod::DeviceConnectionString(cs) => {
+                        info!("Starting provisioning edge device via manual mode using a device connection string...");
+                        let (key_store, provisioning_result, root_key) =
+                            manual_provision_connection_string(&cs, &mut tokio_runtime)?;
+                        start_edgelet!(
+                            key_store,
+                            provisioning_result,
+                            root_key,
+                            force_module_reprovision,
+                            None,
+                        );
+                    }
+                    ManualAuthMethod::X509(x509) => {
+                        info!("Starting provisioning edge device via manual mode using X509 identiy certificate...");
+
+                        let id_data = device_cert_identity_data.ok_or_else(|| {
+                            ErrorKind::Initialize(InitializeErrorReason::DpsProvisioningClient)
+                        })?;
+
+                        let key_bytes = hybrid_identity_key.ok_or_else(|| {
+                            ErrorKind::Initialize(InitializeErrorReason::DpsProvisioningClient)
```

Comment:
>DpsProvisioningClient [](start = 73, length = 21)

nit: This should be ManualProvisioningClient

Reply:
Fixed. Thanks.